### DISCUSSION TITLE
ALE solver

### DISCUSSION THREAD

**Discussion by ehsanakrami** [libMesh/libmesh#3211] [Q&A]
Hi,

I wonder if an ALE solver has been implemented in libMesh. Or any monolithic FSI solver developed using libMesh. It would be appreciated if anyone has developed a monolithic FSI solver and could share it with me.

Thanks

**Comment by roystgnr** [comment]
There's some ALE utilities in the `FEMSystem` tree, but frankly I never got the solver based on it passing tests and it shouldn't be trusted without more work.

There was some work on an FSI solver (using overlapping meshes) going into https://github.com/grinsfem/grins and I think that one was actually in working order, but that project is in mothballs right now so probably also not ready to go out-of-the-box without more work.

There's an FSI solver in https://github.com/idaholab/moose which is actively developed and passing at least its own regression tests on the regular.  That's a massive code, really satisfies your "monolithic" criterion in more ways than one, but might be your best bet.

**Comment by pbauman** [comment]
> I think that one was actually in working order

Sort of. It's a long story that I'm slowly coming back to. It's pretty close. Really depends on getting the finite element space for the fluid right.

> that project is in mothballs right now so probably also not ready to go out-of-the-box without more work

Sadly, this is very true. It's only my spare time and most of that is being eaten up by my kids swim team and dance.
